### PR TITLE
ENH: Add source space SNR estimate

### LINKED ISSUE #1453: SNR in source space
@dengemann pointed me to:

http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2882168/

any interest getting this in MNE-Python?

@mshamalainen any opinion?

### PR DESCRIPTION
**Pull request by klankinen**
- estimate_snr was moved from class BaseSourceEstimate to class SourceEstimate, as it currently works properly only for fixed orientation
- a quick check for fixed/free orientation and warning was also added (see if all the values for etc are > 0)
- G used in calculating b_k2 should not be whitened, otherwise it gives wrong results. This was corrected by taking first output from _prepare_forward function instead of the third one. The correct G to be used is G = G['sol']['data']
- tutorial was updated with examples of MEG-EEG, EEG and MEG according to the reference article

Fixes #1453

### CODE REVIEW

File: mne/source_estimate.py [hunk 1]
```diff
@@ -1505,6 +1505,81 @@ def extract_label_time_course(self, labels, src, mode='mean_flip',
...
+
+        This function should only be used with source estimates with units
+        nanoAmperes (i.e., MNE-like solutions, *not* dSPM or sLORETA).
+
+        .. warning:: This function currently only works properly for fixed
+                     orientation...
+                     
+        Parameters
+        ----------
+        info : instance of io.meas_info.Info
```
Comment (larsoner) [inline mne/source_estimate.py]:
```suggestion
        info : instance Info
```

File: mne/source_estimate.py [hunk 2]
```diff
@@ -1505,6 +1505,81 @@ def extract_label_time_course(self, labels, src, mode='mean_flip',
...
+
+    @verbose
+    def estimate_snr(self, info, fwd, cov, verbose=None):
+        r"""Compute time-varying SNR in the source space.
+
+        This function should only be used with source estimates with units
+        nanoAmperes (i.e., MNE-like solutions, *not* dSPM or sLORETA).
+
+        .. warning:: This function currently only works properly for fixed
+                     orientation...
```
Comment (larsoner) [inline mne/source_estimate.py]: This sentence was meant to be finished, the `...` was a suggestion :)

File: tutorials/source-modeling/plot_source_space_SNR.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+Vaina, L. M., & Stufflebeam, S. M. (2009). Mapping the Signal-To-Noise-Ratios
+of Cortical Sources in Magnetoencephalography and Electroencephalography.
+Human Brain Mapping, 30(4), 1077–1086. doi:10.1002/hbm.20571
+
+"""
+# Author: Padma Sundaram <tottochan@gmail.com>
+#         Kaisu Lankinen <klankinen@mgh.harvard.edu>
+#
+# License: BSD (3-clause)
+
```
Comment (larsoner) [inline tutorials/source-modeling/plot_source_space_SNR.py]: Looks reasonable so far

https://15977-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/source-modeling/plot_source_space_SNR.html

But I would add `# sphinx_gallery_thumbnail_number = 2` up here somewhere to improve the thumbnail

File: mne/source_estimate.py [hunk 4]
```diff
@@ -1505,6 +1505,81 @@ def extract_label_time_course(self, labels, src, mode='mean_flip',
...
+        Parameters
+        ----------
+        info : instance of io.meas_info.Info
+            The measurement info.
+        fwd : instance of Forward
+            The forward solution used to create the source estimate.
+        cov : instance of Covariance
+            The noise covariance used to estimate the resting cortical
+            activations. Should be an evoked covariance, not empty room.
+
```
Comment (larsoner) [inline mne/source_estimate.py]: Azure says you need to add `verbose`

https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=3784&view=logs&jobId=2e1ef983-081d-5c63-b72a-d9f329364ee0

You can do it here with 
```
        %(verbose)s

```
(make sure you maintain a newline between this and `Returns`)

**Comment by larsoner** [comment]
@klankinen I'll push a small commit with some tweaks then this should be good to go

**Comment by larsoner** [comment]
https://16029-1301584-gh.circle-artifacts.com/0/dev/auto_examples/inverse/plot_source_space_SNR.html

**Comment by larsoner** [comment]
Whoops, should have asked for `latest.inc` update. @klankinen @padmasundaram can you give me a URL you want me to add to `names.inc` and I'll make a quick push to master?

**Comment by jasmainak** [comment]
I also realized that there is no test ...

**Comment by agramfort** [comment]
Naughty @larsoner !

**Comment by larsoner** [comment]
Whoops! I'll make a follow-up PR Monday

**Pull request merged**